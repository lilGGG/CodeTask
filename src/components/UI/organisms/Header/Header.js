import Title from "../../atoms/Title/Title";
import SearchItem from "../../molecules/SearchItem/SearchItem";

/**
 * Header component including application title and search input item
 * @param {string} title 
 * @param {string} searchLabel 
 * @param {Method} inputHandler 
 * @returns 
 */
function Header({
  title,searchLabel,inputHandler
}) {
  return (
    <div> 
           <Title Text={title}> </Title>
           <SearchItem text={searchLabel} inputHandler={inputHandler} />
    </div>
  );
}

export default Header;
