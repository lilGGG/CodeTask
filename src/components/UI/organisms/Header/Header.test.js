import React from "react";
import Header from "./Header";
import ReactDom from "react-dom";
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

describe('Header test', () => {
    it('renders snapshot', () => {
      const tree = renderer.create(<Header 
        title="app title" searchLabel="tv" inputHandler={()=> {}}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
  
