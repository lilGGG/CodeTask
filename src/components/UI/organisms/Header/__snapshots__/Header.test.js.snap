// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test renders snapshot 1`] = `
<div>
  <h1
    className="sc-bczRLJ dgaykd"
  >
    app title
  </h1>
  <div>
    <label
      className="sc-dkzDqf kBxcVt"
    >
      tv
    </label>
    <input
      className="sc-gsnTZi dBIVIK"
      onChange={[Function]}
      placeholder="Enter Tv Show Title"
    />
  </div>
</div>
`;
